src/Symfony/Component/Security/Core/User/InMemoryUserProvider.php

```diff
@@ -67,17 +67,17 @@ public function createUser(UserInterface $user)
      */
     public function loadUserByUsername($username)
     {
-        if (!isset($this->users[strtolower($username)])) {
-            $ex = new UsernameNotFoundException(sprintf('Username "%s" does not exist.', $username));
-            $ex->setUsername($username);
-
-            throw $ex;
-        }
-
-        $user = $this->users[strtolower($username)];
+        $storedUser = $this->getUser($username);
```

Comment:
What about keeping the original `$user` variable and keep the next line identical to what it was before.